api_tests/wikis/views/test_wiki_detail.py

```diff
@@ -0,0 +1,170 @@
+from urlparse import urlparse
+from nose.tools import *  # flake8: noqa
+
+from api.base.settings.defaults import API_BASE
+from tests.base import ApiTestCase
+from tests.factories import (AuthUserFactory, ProjectFactory, RegistrationFactory,
+                             NodeWikiFactory, PrivateLinkFactory)
+
+
+class TestWikiDetailView(ApiTestCase):
+    def setUp(self):
+        super(TestWikiDetailView, self).setUp()
+        self.user = AuthUserFactory()
+        self.non_contributor = AuthUserFactory()
+
+    def _add_project_wiki_page(self, node, user):
```

Comment:
These all look very similar. Can you make an ApiWikiTestCase that holds all the setup methods?